src/main/java/seedu/duke/operationlist/PassengerList.java

```diff
@@ -0,0 +1,437 @@
+            resetCheckFormat();
+            throw new SkyControlException(ui.getExceedNameLengthError(name));
+        } else if (isWrongNameFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getNameError());
+        } else if (isWrongDepartureFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getDepartureTimeError());
+        } else if (isWrongFlightNumFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getFlightNumberError());
+        } else if (isWrongBoardingTimeFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getBoardingTimeError());
+        } else if (isWrongGateNumberFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getGateNumberError());
+        } else if (isWrongBoardingGroupFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getBoardingGroupError());
+        } else if (isWrongSeatNumberFormat) {
+            resetCheckFormat();
+            throw new SkyControlException(ui.getSeatNumberError());
+        }
+    }
+
+    //@@author shengiv
+    private void checkPassengerDuplicate() throws SkyControlException {
+        for (int i = 0; i < numOfPassengers; i++) {
+            validatePassenger(i);
+            if (isPassengerDuplicate()) {
+                resetCheckDuplicate();
+                throw new SkyControlException(ui.getDuplicatePassengerError());
+            }
+        }
+    }
+
+    private void modifyPassengersFlightNum(String flightNum, String newFlightNum) {
+        getNumberOfPassengers();
+        for (int i = 0; i < numOfPassengers; i++) {
+            PassengerInfo passenger = passengers.get(i);
+            if (passenger.getFlightNumber().equals(flightNum)) {
+                passenger.setFlightNumber(newFlightNum);
+            }
+        }
+    }
+
+    private void modifyPassengersGateNum(String flightNum, String newGateNum) {
+        getNumberOfPassengers();
+        for (int i = 0; i < numOfPassengers; i++) {
+            PassengerInfo passenger = passengers.get(i);
+            if (passenger.getFlightNumber().equals(flightNum)) {
+                passenger.setGateNumber(newGateNum);
+            }
+        }
+    }
+
+    //@@author ivanthengwr
+    @Override
+    public void deleteOperation(String passengerDetail) throws SkyControlException {
+        getNumberOfPassengers();
+        getPassengerName(passengerDetail);
+        getFlightNumber(passengerDetail);
+        getSeatNumber(passengerDetail);
+        for (int i = 0; i < numOfPassengers; i++) {
+            validatePassenger(i);
+            if (isValidPassenger) {
+                passengers.remove(i);
+                getNumberOfPassengers();
+                Ui.showDeleteMessage(name, flightNumber, seatNumber, numOfPassengers);
+                isDeleteSuccess = true;
+                break;
```

Comment:
Varying levels of abstraction, consider adding more SLAP to achieve single level abstraction.